### PR TITLE
S3t6 update-win-count route

### PR DESCRIPTION
**Pull request by MotNosniktaw**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -41,3 +41,42 @@ app.get('/dogs', (req, res) => {
...
+            success: false,
+            message: 'Please ensure request has valid "winnerID" and "winCount" and try again.',
+            data: []
+        })
+    }
+    mongoClient.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true }, (err, client) => {
+        if (err) {
+            return res.status(500).json({
+                success: false,
+                message: 'Error conencting to database.',
```
Comment (FelixLetheren) [inline index.js]: spelling

Comment (lizsharp) [inline index.js]: bit of a typo here - Error connecting to database

File: index.js [hunk 2]
```diff
@@ -41,3 +41,52 @@ app.get('/dogs', (req, res) => {
...
+                message: 'Not able to update database',
+                data: []
+            })
+        }
+    })
+})
+
+const declareChampion = function (db, details, callback) {
+    var collection = db.collection(collectionName)
+    newCount = details.winCount + 1
```
Comment (AlisonYoon) [inline index.js]: let or var or something

File: index.js [hunk 3]
```diff
@@ -41,3 +41,61 @@ app.get('/dogs', (req, res) => {
...
+            {$set: {"winCount": newCount}},
+            function (err, result) {
+                console.log('Database has been updated')
+                callback(result)
+        })
+    }
+    catch (err) {
+        console.log(err)
+    }
+
```
Comment (AlisonYoon) [inline index.js]: remove newline

File: index.js [hunk 4]
```diff
@@ -41,3 +41,61 @@ app.get('/dogs', (req, res) => {
...
+    let collection = db.collection(collectionName)
+    try {
+        let winner = await collection.find({"_id": ObjectId(details.winnerID)}).toArray()
+        if (!winner[0].hasOwnProperty('winCount')) {
+            winner[0].winCount = 0
+        }
+        let newCount = parseInt(winner[0].winCount) + 1
+        collection.updateOne(
+            {"_id": ObjectId(winner[0]._id)},
+            {$set: {"winCount": newCount}},
```
Comment (mallowmew) [inline index.js]: You could try using `$inc` to increment the wincount in a single operation. https://docs.mongodb.com/manual/reference/operator/update/inc/#up._S_inc

**Review by mallowmew — APPROVED** [review]

File: index.js (hunk 3 above)
Comment (MotNosniktaw) [reply index.js]: done

File: index.js (hunk 2 above)
Comment (MotNosniktaw) [reply index.js]: done

File: index.js (hunk 1 above)
Comment (MotNosniktaw) [reply index.js]: done

Comment (MotNosniktaw) [reply index.js]: done

**Review by FelixLetheren — APPROVED** [review]

**Pull request merged**